File: nautilus-server/src/main/java/co/kaioru/nautilus/server/ICluster.java
package co.kaioru.nautilus.server;

import co.kaioru.nautilus.core.config.IConfigurableRemote;
import co.kaioru.nautilus.server.config.ClusterConfig;

import java.rmi.RemoteException;
import java.util.Collection;

public interface ICluster<S extends IShard, CO extends ClusterConfig> extends IConfigurableRemote<CO>, IDaemon<CO> {

	Collection<? extends S> getShards() throws RemoteException;

	void registerShard(S shard) throws RemoteException;

	void deregisterShard(S shard) throws RemoteException;

}

File: nautilus-server/src/main/kotlin/co/kaioru/nautilus/interop/ICluster.kt
package co.kaioru.nautilus.interop

import co.kaioru.nautilus.interop.info.ClusterInfo
import java.rmi.RemoteException

interface ICluster<out I : ClusterInfo> : INode<I> {
	val shards: MutableCollection<IShard<*, *>>
		@Throws(RemoteException::class) get

	@Throws(RemoteException::class)
	fun registerShard(shard: IShard<*, *>)

	@Throws(RemoteException::class)
	fun deregisterShard(shard: IShard<*, *>)
}
